### PR TITLE
feature/oh-my-zsh-completions

### PR DESCRIPTION
**Pull request by noameron**
Add Zsh Shell Completions

  Adds tab completion support for the OpenSpec CLI with Zsh (including Oh My Zsh).

  New Command

  openspec completion install    # Auto-installs Zsh completions
  openspec completion uninstall  # Removes completions
  openspec completion generate   # Outputs completion script

  Key Files

  Core Implementation
  - src/core/completions/command-registry.ts - Centralized registry of all CLI commands and flags with COMMON_FLAGS constant for reusable flag definitions
  - src/core/completions/completion-provider.ts - Discovers change/spec IDs from projects with 2s caching
  - src/core/completions/factory.ts - Factory pattern with SUPPORTED_SHELLS constant for shell support

  Zsh-Specific
  - src/core/completions/generators/zsh-generator.ts - Generates Zsh completion scripts using _arguments and _describe
  - src/core/completions/installers/zsh-installer.ts - Handles installation to Oh My Zsh (~/.oh-my-zsh/custom/completions/) or standard Zsh (~/.zsh/completions/)

  Command Interface
  - src/commands/completion.ts - CLI command implementation with shared resolveShellOrExit() logic

  Features

  - Tab complete commands, flags, change IDs, and spec IDs
  - Auto-detects Oh My Zsh vs standard Zsh
  - Only activates in OpenSpec projects
  - Plugin architecture for future shells (bash, fish, PowerShell)

## Summary by CodeRabbit

* **New Features**
  * Adds a top-level "completion" CLI with generate/install/uninstall for Zsh; dynamic, project-aware suggestions for changes/specs (2s cache); auto shell detection; machine-readable mode; Oh My Zsh & standard Zsh install/uninstall flows.

* **Documentation**
  * Adds detailed design, proposal, and spec covering architecture, UX, install/uninstall behavior, error messages, caching, testing, and roadmap for other shells.

* **Tests**
  * Extensive tests for generation, installer flows, detection, caching, and edge cases.

* **Chores**
  * Postinstall auto-install wiring with opt-out and a test harness.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/commands/completion.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    this.completionProvider = new CompletionProvider();
+  }
+  /**
+   * Resolve shell parameter or exit with error
+   *
+   * @param shell - The shell parameter (may be undefined)
+   * @param operationName - Name of the operation (for error messages)
+   * @returns Resolved shell or null if should exit
+   */
+  private resolveShellOrExit(shell: string | undefined, operationName: string): SupportedShell | null {
```
Comment (TabishB) [inline src/commands/completion.ts:43]: The spec mentions that users using non-supported shells should respond with the following:
> - **THEN** throw error: "Shell '<name>' is not supported yet. Currently supported: zsh"

But we're responding here with "Could not auto-detect shell". 

I think what the spec mentions is a better approach of detected-but-unsupported

File: src/core/completions/installers/zsh-installer.ts [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+  /**
+   * Generate .zshrc configuration content
+   *
+   * @param completionsDir - Directory containing completion scripts
+   * @returns Configuration content
+   */
+  private generateZshrcConfig(completionsDir: string): string {
+    return [
+      '# OpenSpec shell completions configuration',
+      `fpath=(${completionsDir} $fpath)`,
```
Comment (TabishB) [inline src/core/completions/installers/zsh-installer.ts]: Do we need to wrap `${completionsDir}` in quotes here? Right now the generated line is `fpath=(/Users/Jane Doe/.zsh/completions $fpath)` for anyone with spaces in their home path, which Zsh splits into multiple entries.

Updating this to `fpath=("${completionsDir}" $fpath)` (or escaping spaces) would keep the directory intact so compinit can find the file.

File: src/core/completions/installers/zsh-installer.ts [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+
+  /**
+   * Install the completion script
+   *
+   * @param completionScript - The completion script content to install
+   * @returns Installation result with status and instructions
+   */
+  async install(completionScript: string): Promise<InstallationResult> {
+    try {
+      const { path: targetPath, isOhMyZsh } = await this.getInstallationPath();
```
Comment (TabishB) [inline src/core/completions/installers/zsh-installer.ts:266]: For Oh My Zsh installs we never verify that custom/completions is actually in `$fpath`; we just drop the file and print guidance. 

The spec’s “Installing for Oh My Zsh” scenario (…spec.md:103-110) explicitly says to ensure the directory is added to ~/.zshrc when needed. 

Could we inspect/update .zshrc (or otherwise confirm $fpath contains the path) instead of leaving it to the user?

File: src/core/completions/installers/zsh-installer.ts [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+
+      // Ensure the directory exists
+      const targetDir = path.dirname(targetPath);
+      await fs.mkdir(targetDir, { recursive: true });
+
+      // Backup existing file if present
```
Comment (TabishB) [inline src/core/completions/installers/zsh-installer.ts]: We always overwrite `_openspec` and report success even if the script was already installed. Spec calls for detecting the “already installed” case and letting the user know/offer reinstall. 

Can we check for an existing file (and maybe diff) before writing, and emit the spec-mandated messaging?

**Comment by noameron** [comment]
@TabishB Hi, this PR is abit big partly due to the specs used

File: src/core/completions/installers/zsh-installer.ts (hunk 4 above)
Comment (noameron) [reply src/core/completions/installers/zsh-installer.ts]: @TabishB Do you prefer prompting the user if the file exists but has different setting that needs to be made? or should just alert the user file exists and what he changed? pure visiblity

Comment (TabishB) [reply src/core/completions/installers/zsh-installer.ts]: Pure visibility is always better IMO. Would avoid interactive option, don't think we need it.

**Review by TabishB — APPROVED** [review]
Thanks @noameron for the contribution 🚀
